packages/server-wallet/src/models/channel.ts

```diff
@@ -325,39 +326,40 @@ export class Channel extends Model implements RequiredColumns {
   }
 }
 
-export enum Errors {
-  invalidChannelId = 'Invalid channel id',
-  incorrectHash = 'Incorrect hash',
-  duplicateTurnNums = 'multiple states with same turn number',
-  notSorted = 'states not sorted',
-  multipleSignedStates = 'Store signed multiple states for a single turn',
-  staleState = 'Attempting to sign a stale state',
-  channelMissing = 'No channel found with id.',
-  channelFunded = 'Channel already funded.',
-  channelLocked = 'Channel is locked',
-  noBudget = 'No budget exists for domain. ',
-  noAssetBudget = "This domain's budget does contain this asset",
-  channelNotInBudget = "This domain's budget does not reference this channel",
-  noDomainForChannel = 'No domain defined for channel',
-  domainExistsOnChannel = 'Channel already has a domain.',
-  budgetAlreadyExists = 'There already exists a budget for this domain',
-  budgetInsufficient = 'Budget insufficient to reserve funds',
-  amountUnauthorized = 'Amount unauthorized in current budget',
-  cannotFindDestination = 'Cannot find destination for participant',
-  cannotFindPrivateKey = 'Private key missing for your address',
-  notInChannel = 'Attempting to initialize  channel as a non-participant',
-  noLedger = 'No ledger exists with peer',
-  amountNotFound = 'Cannot find allocation entry with destination',
-  invalidNonce = 'Invalid nonce',
-  invalidTransition = 'Invalid transition',
-  invalidAppData = 'Invalid app data',
-  emittingDuringTransaction = 'Attempting to emit event during transaction',
-  notMyTurn = "Cannot update channel unless it's your turn",
-}
-
-class ChannelError extends Error {
-  readonly type = 'ChannelError';
-  constructor(reason: Errors, public readonly data: any = undefined) {
+export class ChannelError extends WalletError {
+  readonly type = WalletError.errors.ChannelError;
+
+  static readonly reasons = {
```

Comment:
Should we take the opportunity to prune these at all?

Reply:
Good call: 3a8f3a169